src/src/components/Charts/TimeChart.tsx

```diff
@@ -0,0 +1,310 @@
+import _default from 'chart.js/dist/plugins/plugin.tooltip';
+import React from 'react';
+
+import { useSimpleStorage } from '../../hooks/useStorage';
+import { commonText } from '../../localization/common';
+import type { IR, RA } from '../../utils/types';
+import { Button } from '../Atoms';
+import { DAY, formatNumber, HOUR, MINUTE } from '../Atoms/Internationalization';
+import type { CalendarListEntry } from '../Contexts/CalendarsContext';
+import { CalendarsContext } from '../Contexts/CalendarsContext';
+import type { DayHours, EventsStore } from '../EventsStore';
+import { summedDurations } from '../EventsStore';
+import { CalendarIndicator } from '../Molecules/CalendarIndicator';
+import { WidgetContainer } from '../Widgets/WidgetContainer';
+
+export type TimeChartMode = 'average' | 'total';
+const stickyColumn = 'sticky left-0 bg-white';
+const darkened = 'bg-gray-300';
+const extraDarkened = 'bg-gray-400';
+
+export function TimeChart({
+  label,
+  durations,
+}: {
+  readonly label: string;
+  readonly durations: EventsStore | undefined;
+}): JSX.Element {
+  const [mode, setMode] = useSimpleStorage('timeChartMode', 'total');
+
+  const calendars = React.useContext(CalendarsContext)!;
+  const times = React.useMemo(
+    () =>
+      durations === undefined || mode === undefined
+        ? undefined
+        : getTimes(durations, mode),
+    [durations, mode]
+  );
+
+  return (
+    <WidgetContainer
+      buttons={
+        <div className="flex gap-2">
+          <span className="sr-only">{commonText('viewMode')}</span>
+          <Button.White
+            aria-pressed={mode === 'total' ? true : undefined}
+            title={commonText('viewMode')}
+            onClick={(): void => setMode('total')}
+          >
+            {commonText('totalHours')}
+          </Button.White>
+          <Button.White
+            aria-pressed={mode === 'average' ? true : undefined}
+            title={commonText('viewMode')}
+            onClick={(): void => setMode('average')}
+          >
+            {commonText('averageMinutes')}
+          </Button.White>
+        </div>
+      }
+      header={label}
+    >
+      {durations === undefined || times === undefined || mode === undefined ? (
+        commonText('loading')
+      ) : (
+        <table
+          className={`
+            grid-table grid-cols-[auto_repeat(24,1fr)_1fr]
+            overflow-auto [&_:is(th,td)]:p-2
+            [&_td]:justify-center
+          `}
+        >
+          <thead>
+            <tr>
+              <th className={stickyColumn} scope="col">
+                <span className="sr-only">{commonText('calendars')}</span>
+              </th>
+              {Array.from({ length: DAY / HOUR }, (_, index) => (
+                <th className="justify-center" key={index} scope="col">
+                  {/* FIXME: display AM/PM when appropriate */}
```

Comment:
Are we creating a future work item for this?

Reply:
Good catch. Yes, created https://github.com/maxxxxxdlp/calendar-plus/issues/179 for it